### ISSUE TITLE
Converts variables with @$ to _at_$

### ISSUE THREAD

**Issue by tjfleming** [wearefractal/gulp-coffee#56]
I have some AngularJS modules written in coffeescript.  They suddenly stopped working with error message like "_at_$stateProvider" not found.  I compare the compiled javascript between gulp-coffee 2.2.0 and 2.2.1 and see that wherever I refer to a variable or parameter with @$, it gets converted to _at_$.

**Comment by contra** [comment]
Can you try it with the latest version?

**Comment by tjfleming** [comment]
No change.

`Error: [$injector:unpr] Unknown provider: _at_$stateProvider <- _at_$state <- HeaderCtrl`

I've changed my code so I don't pass `@$` anymore so it isn't a problem for me.

**Comment by contra** [comment]
Can you post your task where this is happening?

**Comment by leonardfactory** [comment]
Same error here. My gulp task is:

```coffee
gulp.task 'dev:app', ->
  gulp.src paths.src
    .pipe plumber onError
    .pipe sourcemaps.init()
    .pipe coffee bare: true
    .pipe angularFilesort()
    .pipe concat 'app.js'
    .pipe ngAnnotate()
    .pipe sourcemaps.write()
    .pipe gulp.dest paths.dist + '/js'
```

A simple controller with this error is the following (I'm using ngAnnotate):

```coffee
angular
  .module 'test'
  .controller 'MainController',
    class MainController

      ### @ngInject ###
      constructor: (@$scope) ->
        @$scope.title = 'Hello'
```

The parameter is converted into `_at_$scope`

**Comment by contra** [comment]
Can you isolate this in a repo that reproduces the bug?

**Comment by bcherny** [comment]
This was caused by https://github.com/jashkenas/coffeescript/issues/3816

It was fixed in 1.9.1, see http://coffeescript.org/#1.9.1
